src/main/flight/navigation_rewrite.c

```diff
@@ -951,16 +951,11 @@ static navigationFSMEvent_t navOnEnteringState_NAV_STATE_RTH_3D_LANDING(navigati
             updateAltitudeTargetFromClimbRate(-0.15f * posControl.navConfig->land_descent_rate, CLIMB_RATE_RESET_SURFACE_TARGET);
         }
         else {
-            // Gradually reduce descent speed depending on actual altitude.
-            if (posControl.actualState.pos.V.Z > (posControl.homePosition.pos.V.Z + 1500.0f)) {
-                updateAltitudeTargetFromClimbRate(-1.0f * posControl.navConfig->land_descent_rate, CLIMB_RATE_RESET_SURFACE_TARGET);
-            }
-            else if (posControl.actualState.pos.V.Z > (posControl.homePosition.pos.V.Z + 500.0f)) {
-                updateAltitudeTargetFromClimbRate(-0.5f * posControl.navConfig->land_descent_rate, CLIMB_RATE_RESET_SURFACE_TARGET);
-            }
-            else {
-                updateAltitudeTargetFromClimbRate(-0.25f * posControl.navConfig->land_descent_rate, CLIMB_RATE_RESET_SURFACE_TARGET);
-            }
+            // Ramp down decend velocity from 100% at 2000cm altitude to 25% from 500cm to 0cm.
+            // TODO: make the number 2000.0f a setting, call it something like rth_decendvel_rampheight.
+            float decendVelScaling = (posControl.actualState.pos.V.Z - posControl.homePosition.pos.V.Z) / 2000.0f;
+            decendVelScaling = constrainf(decendVelScaling, 0.25f, 1.0f);
```

Comment:
I would make that 2000 two settings actually - something like `nav_land_slowdown_maxalt` and `nav_land_slowdown_minalt` and scale from 100% to 25% between them. Something like:
```
decendVelScaling = (posControl.actualState.pos.V.Z - posControl.homePosition.pos.V.Z - 500.0f) / (2000.0f - 500.0f) * 0.75f + 0.25f;  // Yield 1.0 at 2000 alt and 0.25 at 500 alt
decendVelScaling = constrainf(decendVelScaling, 0.25f, 1.0f);
 ```